### PR TITLE
[server] reduce build time

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

Include the db tests in the regular `yarn test` script.
Also, removes `prepare` script from package.json as it is automatically executed post `yarn install` and duplicates the `yarn clean && yarn build`.

## Related Issue(s)

Fixes #

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/server/package.json [hunk 1]
```diff
@@ -16,11 +15,11 @@
     "clean:node": "rimraf node_modules",
     "purge": "yarn clean && yarn clean:node && yarn run rimraf yarn.lock",
     "lint": "yarn tslint -p .",
-    "test": ". $(leeway run components/gitpod-db:db-test-env) && TS_NODE_FILES=true mocha --opts mocha.opts './**/*.spec.ts' --exclude './node_modules/**'",
-    "spicedb-testing": "if netstat -tuln | grep ':50051 '; then echo '`spicedb serve-testing` is already running.'; else spicedb serve-testing --load-configs /workspace/gitpod/install/installer/pkg/components/spicedb/data/schema.yaml; fi",
-    "stop-spicedb-testing": "PID=$(lsof -t -i:50051); if [ -z \"$PID\" ]; then echo '`spicedb serve-testing` is not running'; else kill -INT $PID && echo 'Stopped `spicedb serve-testing`'; fi",
-    "db-test": "r(){ . $(leeway run components/gitpod-db:db-test-env); yarn spicedb-testing & yarn db-test-run; yarn stop-spicedb-testing; };r",
-    "db-test-run": "TS_NODE_FILES=true mocha --opts mocha.opts '**/*.spec.db.ts' --exclude './node_modules/**'",
+    "test": ". $(leeway run components/gitpod-db:db-test-env) && yarn test:unit && (yarn start-spicedb & (yarn test:db && yarn stop-spicedb))",
```
Comment (geropl) [inline components/server/package.json:18]: I assume this works because `yarn test` is the default `command.test` for yarn packages? :thinking:

Comment (svenefftinge) [reply components/server/package.json:18]: Yes, not cool that this is hidden within leeway. I always need to look up the leeway code in order to understand the semantics.

Comment (geropl) [reply components/server/package.json:18]: :100: Would it make sense to explicitly add that to the package? config.commands.test is the path I think.

**Review by geropl — APPROVED** [review]
Code LGTM, build works! :heavy_check_mark: 

I still see a stray `tslint -p` in the logs, but we can find that later.

**Comment by svenefftinge** [comment]
> I still see a stray tslint -p in the logs, but we can find that later.

this wasn't rebased on the other eslint PR yet.

**Pull request merged**